fix: Add chaijs/get-func-name resolution

leotm opened the pull request:
## **Description**

Fix CI: https://github.com/MetaMask/metamask-mobile/actions/runs/6331863392/job/17197486695
By adding a Yarn <s>package.json</s> lockfile resolution

Then once
- https://github.com/ethereum-optimism/optimism/pull/7432

Merged we can remove the resolution and bump `ethereum-optimism`

Fixed in: https://github.com/chaijs/get-func-name/commit/f934b228b5e2cb94d6c8576d3aac05493f667c69

## **Manual testing steps**

`yarn audit:ci`

## **Related issues**

_Fixes https://github.com/chaijs/get-func-name/security/advisories/GHSA-4q6p-r6v2-jvc5_

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [x] I’ve indicated what issue this PR is linked to: Fixes #???
- [ ] I’ve included tests if applicable.
- [ ] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

leotm committed:
Fix GHSA-4q6p-r6v2-jvc5

Follow-up: Bump ethereum-optimism then remove resolution
Ref: https://github.com/ethereum-optimism/optimism/pull/7432#pullrequestreview-1648545367

Vulnerability: https://github.com/chaijs/get-func-name/security/advisories/GHSA-4q6p-r6v2-jvc5

Fix commit: https://github.com/chaijs/get-func-name/commit/f934b228b5e2cb94d6c8576d3aac05493f667c69

tommasini reviewed (dismissed):
LGTM!

Gudahtt commented on package.json:
```diff
@@ -148,7 +148,8 @@
     "**/fast-xml-parser": "4.2.4",
-    "tough-cookie": "4.1.3"
+    "tough-cookie": "4.1.3",
+    "**/get-func-name": "2.0.2"
```
It looks like the lockfile update is all we need here, since the update is within the range being asked for already. Could we remove this resolution entry? Better to minimize resolutions if possible because they could result in incompatible versions being used in future updates

leotm replied:
great point ^ more minimal and better minimise our resolutions

https://github.com/MetaMask/metamask-mobile/pull/7351#event-10498442076

leotm committed:
Remove package.json resolution entry

RE: https://github.com/MetaMask/metamask-mobile/pull/7351#discussion_r1340002910

Gudahtt reviewed (dismissed):
LGTM!

leotm commented:
@SocketSecurity ignore get-func-name@2.0.2

nb: https://socket.dev/npm/package/get-func-name/issues/2.0.2?issue=unstableOwnership

> A new collaborator has begun publishing package versions. Package stability and security risk may be elevated. Found 1 instance in 1 package

leotm commented on yarn.lock:
```diff
@@ -17377,10 +17377,10 @@ get-caller-file@^2.0.1, get-caller-file@^2.0.5:
-  version "2.0.0"
-  resolved "https://registry.yarnpkg.com/get-func-name/-/get-func-name-2.0.0.tgz#ead774abee72e20409433a066366023dd6887a41"
-  integrity sha512-Hm0ixYtaSZ/V7C8FJrtZIuBBI+iSgL+1Aq82zSu8VQNB4S3Gk8e7Qs3VwBDJAhmRZcFqkl3tQu36g/Foh5I5ig==
+get-func-name@2.0.2, get-func-name@^2.0.0:
```
> I thought we had protection from stale lockfiles

for some reason expected Yarn to do this automatically 😅

leotm replied:
https://github.com/MetaMask/metamask-mobile/pull/7351/commits/11adc800309fd05ebca572165409940f1cb5981f

leotm committed:
Update yarn.lock

Remove stale lockfile resolution
Since removed package.json resolution

RE: https://github.com/MetaMask/metamask-mobile/pull/7351/files#r1340055974

leotm commented:
ah automation wins, could've merged instead 
- https://github.com/MetaMask/metamask-mobile/pull/7347

🙈 @dawnseeker8 i'll lookout for our bots next time
